### PR TITLE
Use OpenTelemetry SDK in HotROD 🚗

### LINKED ISSUE #3380: Upgrade HotROD example to use OpenTelemetry SDK
Per #3362

* [x] replace Jaeger SDK with OTEL SDK using an OT Bridge 
* [x] replace OT instrumentation with OpenTelemetry
* [x] Remove references to OpenTracing API
  * [x] from HotROD code
  * [x] from HotROD tests `examples/hotrod/pkg/tracing/rpcmetrics/`
  * [x] from README #4790
* [x] Remove Jaeger exporter, leave only OTLP, and update README and docker-compose

### PR DESCRIPTION
**Pull request by yurishkuro**
Based on earlier PR #3390 by @rbroggi. 

## Which problem is this PR solving?
- Resolves #3380

## Short description of the changes
- Switch from jaeger-client-go SDK to OTel SDK paired with ot-otel bridge
- Add cli flag to select which Otel Exporter to use (Jaeger, OTLP or stdout)
- Add favicon 🚗
- Added OTEL version of rpcmetrics from jaeger-client-go
- Remove dependency on github.com/uber/jaeger-lib in HotROD (addresses one of outstanding tasks in #3766)

### CODE REVIEW

File: examples/hotrod/pkg/tracing/init.go [hunk 1]
```diff
@@ -16,56 +16,69 @@
...
 
-	_, err := cfg.FromEnv()
-	if err != nil {
-		logger.Bg().Fatal("cannot parse Jaeger env vars", zap.Error(err))
-	}
+// Init initializes OpenTelemetry SDK and uses OTel-OpenTracing Bridge
+// to return an OpenTracing-compatible tracer.
+func Init(serviceName string, exporterType string, logger log.Factory) opentracing.Tracer {
+	once.Do(func() {
+		otel.SetTextMapPropagator(propagation.TraceContext{})
```
Comment (yurishkuro) [inline examples/hotrod/pkg/tracing/init.go:45]: @rbroggi your PR was mostly spot-on, this was the only thing missing, so traces did not continue across services.

**Review by albertteoh — APPROVED** [review]
LGTM; do you think this should be the approach (i.e. using the OTEL -> OT shim) to take for #3381?

File: examples/hotrod/pkg/tracing/init.go [hunk 2]
```diff
@@ -16,56 +16,69 @@
...
-}
-
-type jaegerLoggerAdapter struct {
-	logger log.Logger
-}
 
-func (l jaegerLoggerAdapter) Error(msg string) {
-	l.logger.Error(msg)
+	tp := sdktrace.NewTracerProvider(
+		sdktrace.WithBatcher(exp, sdktrace.WithMaxExportBatchSize(1)),
```
[the remark this answers is not in the document]
Comment (yurishkuro) [reply examples/hotrod/pkg/tracing/init.go:54]: or, this should be removed, I used it when troubleshooting.

**Comment by yurishkuro** [comment]
@albertteoh my intention was to make this change work with the original [HotROD blog post](https://medium.com/opentracing/take-opentracing-for-a-hotrod-ride-f6e3141f7941) (probably time to update it to current state). But it has a section that demonstrates generation of RPC metrics by using span observer in the tracer. This functionality is now broken, but I suspect it's possible to reproduce it via an additional span processor passed to the OTEL SDK

**Comment by yurishkuro** [comment]
> do you think this should be the approach (i.e. using the OTEL -> OT shim) to take for https://github.com/jaegertracing/jaeger/issues/3381?

I don't have a strong opinion there, but it's probably better to just do a full upgrade to OTEL. I kinda wanted to try out the OT-OTEL shim in the Hotrod first, since I haven't seen any pubs describing a successful use of the shim on existing code. Not that I know it works (maybe write a blog post about it),  it may be worth doing a follow-up PR and move to OTEL directly. It will involve using different instrumentation libraries though.

File: examples/hotrod/pkg/tracing/init.go (hunk 1 above)
Comment (rbroggi) [reply examples/hotrod/pkg/tracing/init.go:47]: Thx for reaching out. I have been quite in the shadows lately. Hope to come back to some contributions soon.

**Pull request merged**